src/commands/deployments/editScmType.ts

```diff
@@ -18,7 +18,11 @@ export async function editScmType(context: IActionContext, node?: SiteTreeItem |
         node = <SiteTreeItem>node.parent;
     }
 
-    await appservice.editScmType(node.root.client, node, context, newScmType, showToast);
+    if (node instanceof DeploymentsTreeItem && node.parent instanceof AzureParentTreeItem) {
+        await appservice.editScmType(context, node.root.client, node.root, newScmType, showToast);
+    } else if (node instanceof SiteTreeItem && node.deploymentsNode) {
```

Comment:
You'll have to do a refresh on the node if the `deploymentsNode` doesn't exist.  If you make an async create method, you won't have to do this check.

Right now, if you try to call this from the command palette, it just doesn't do anything.